### PR TITLE
Test schema corrections

### PR DESCRIPTION
**Pull request by patrickkwang**
This includes assorted fixes to bring the "simple" test schema in line with OpenAPI 3.0.

These issues were identified by attempting to validate the schema against [the OpenAPI schema](https://github.com/OAI/OpenAPI-Specification/blob/master/schemas/v3.0/schema.json) using [jsonschema](https://github.com/Julian/jsonschema).

### CODE REVIEW

File: tests/fixtures/simple/openapi.yaml [hunk 1]
```diff
@@ -1021,10 +1039,10 @@ components:
       required:
         - image_version
   parameters:
-    OptionalName:
+    Name:
       name: name
       in: path
       description: Name of the person to greet.
-      required: false
```
Comment (dtkav) [inline tests/fixtures/simple/openapi.yaml]: All of the other changes make sense to me (adding descriptions, clarifying style and explode, etc.
But this is a change in the actual content and intent of the endpoint. Can you say more about why you did this?

**Comment by dtkav** [comment]
Thanks for doing this. I'm excited to be able to use the OpenAPI schema for validation.
I left a minor comment, otherwise looking good!

File: tests/fixtures/simple/openapi.yaml (hunk 1 above)
Comment (patrickkwang) [reply tests/fixtures/simple/openapi.yaml]: Per the OpenAPI specification, path parameters _must_ have `required: true`. The nominal purpose of this test is to "check if add_operation is called only on http methods field", which does not obviously require that the name parameter by optional, so it seemed reasonable to make the name required. But perhaps there is some deeper purpose that I'm missing.

**Comment by hjacobs** [comment]
Could you rebase this PR?

**Comment by dtkav** [comment]
Hey @patrickkwang - If you give me write permissions to the PR I can resolve the conflict.
I'd love to get this and #936 through.

File: tests/fixtures/simple/openapi.yaml (hunk 1 above)
Comment (dtkav) [reply tests/fixtures/simple/openapi.yaml]: ah yep - I missed that it was a path param. :+1:

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**